### DISCUSSION TITLE
Akvorado doesn't show AS-Path from BMP

### DISCUSSION THREAD

**Discussion by bytedisaster** [akvorado/akvorado#1369] [Q&A]
Hello! Thank you for your product!

Recently, I have enabled BMP to collect routing data from my border routers (Huawei NE 8000). 
Generally, it's working. But Akvorado aren't showing me dst as-path in it's web interface and I'm not able to use this attribute in filters. But since it shows dst community I believe that there's no problem with BMP session itself. 
I took a traffic dump and saw that as-path is present in update messages. 
What I see in Akvorado:
DstAS	198758
DstASPath	[]
DstCommunities	[ "31133:1000", "31133:1023" ]

What can I check to make shure what's the problem?

Thaks a lot!

**Comment by vincentbernat** [comment]
Are you using sFlow? If there is an extended gateway record, AS path is taken from here.

**Reply by bytedisaster** [reply]
Yes, I am. 
But what about communities?

**Reply by vincentbernat** [reply]
Is sFlow sending an extended gateway record? If yes, even when the as path is empty, Akvorado considers this is the correct one. I suppose this could be changed, but an empty AS path is a valid one.

**Reply by bytedisaster** [reply]
By your mention I've checked sflow data.
Yes, it were sending extended route data, as well as 'extended gateway data'. They both contain similar attributes. Anyway, I found the way to disable sending 'extended route data'. Now it doesn't send 'extended route data' dataset. But it still does send 'extended gateway data'. Unfortually, there's no way to disable this on the router. 

Do I understand right, that data from sflow has priority before data from BMP? And if Akvorado are receiveng some route data from sflow, it's ignoring data from BMP? If so, is there a way to change this behavior?

**Reply by bytedisaster** [reply]
Both of these data sets do not contain as-path attribute

**Reply by vincentbernat** [reply]
Yes, you are correct, once you have the extended gateway data, Akvorado won't take AS path from BMP. I have fixed that in commit 1f7471a92670.

**Reply by bytedisaster** [reply]
Thank you! Will be waiting for an update.
